File: Flyweight/Golang/PlayerFactory.go
package main

type PlayerFactory struct {
	players map[string]IPlayer
}

func (playerFactory *PlayerFactory) createPlayer(playerType string) IPlayer {
	if playerFactory.players == nil {
		playerFactory.players = make(map[string]IPlayer)
	}

	if val, ok := playerFactory.players[playerType]; ok {
		return val
	} else {
		var player IPlayer		

		switch playerType {
			case "Terrorist":
				player = &Terrorist{}
				playerFactory.players[playerType] = player

				return player
				break
			
			case "CounterTerrorist":
				player = &CounterTerrorist{}
				playerFactory.players[playerType] = player

				return player
				break
		}
	}
	return nil
}

File: Flyweight/Dart/PlayerFactory.dart
import 'CounterTerrorist.dart';
import 'IPlayer.dart';
import 'Terrorist.dart';

class PlayerFactory {
  Map _players = {};

  IPlayer? createPlayer(String type) {
    if (_players[type] != null) {
      return _players[type];
    } else {
      IPlayer? player = null;

      switch (type) {
        case "Terrorist":
          player = Terrorist();
          _players[type] = player;
          break;
        case "CounterTerrorist":
          player = CounterTerrorist();
          _players[type] = player;
          break;
      }

      return player;
    }
  }
}
